docs/metrics.md

```diff
@@ -4,30 +4,173 @@ Let's look at the various metrics you can calculate using Valor.
+| :- | :- | :- |
+| Intersection Over Union (IoU) | A ratio between the groundtruth and predicted regions of an image, measured as a percentage, grouped by class. |$$\dfrac{area( prediction \cap groundtruth )}{area( prediction \cup groundtruth )}$$ |
+| Mean IoU 	| The average of IoUs, calculated over several different classes. | $$\dfrac{1}{\|classes\|} \sum\limits_{c \in classes} IoU_{c}$$ |
+
+# Methods
+
+## ROC AUC for Classification
+
+### Determining Binary Truth
+
+| Element | Description |
+| ------- | ------------ |
+| True Positive (TP) | Prediction confidence is above a threshold and is correct. |
+| False Positive (FP) | Prediction confidence is above a threshold and is incorrect. |
+| True Negative (TN) | Prediction confidence is below a threshold and is correct. |
+| False Negative (FN) | Prediction confidence is below a threshold and is incorrect. |
+
+- $\text{True Positive Rate (TPR)} = \dfrac{|TP|}{|TP| + |FN|}$
+
+- $\text{False Positive Rate (FPR)} = \dfrac{|FP|}{|FP| + |TN|}$
+
+### Receiver Operating Characteristic (ROC)
+
+An ROC curve plots TPR vs. FPR at different confidence thresholds.
+
+### Area under the ROC curve (ROC AUC)
+
+From the ROC curve we calculate the ROC AUC metric by taking the integral using the trapezoidal rule formula.
+
+$$
+\sum_{i=1}^{|scores|} \frac{(TPR(scores_{i-1}) - TPR(scores_i)) \cdot (FPR(scores_{i-1}) - FPR(scores_i))}{2}
+$$
+
+### References
+- [Classification: ROC Curve and AUC](https://developers.google.com/machine-learning/crash-course/classification/roc-and-auc)
+
+## Average Precision (AP) for Object Detection
+
+For object-detection and instance segmentation tasks, average-precision is calculated from the intersection-over-union (IoU) of geometric annotations.
+
+### What is Intersection-over-Union (IoU)?
+
+IoU is the ratio of intersecting area over total area spanned by two geometries.
+
+$$Intersection \ over \ Union \ (IoU) = \dfrac{Area( prediction \cap groundtruth )}{Area( prediction \cup groundtruth )}$$
+
+### Multiclass Precision and Recall
+
+| Case | Description |
+| :- | :- |
+| True Positive (TP) | Prediction meets IoU threshold requirements. |
+| False Positive (FP) | Prediction fails IoU threshold requirements. |
+| True Negative (TN) | Unused in multi-class evaluation.
+| False Negative (FN) | No prediction exists. |
+
+- $Precision = \dfrac{|TP|}{|TP| + |FP|} = \dfrac{\text{Number of True Predictions}}{|\text{Predictions}|}$
+
+- $Recall = \dfrac{|TP|}{|TP| + |FN|} = \dfrac{\text{Number of True Predictions}}{|\text{Groundtruths}|}$
+
+### Finding the best prediction for a groundtruth.
+
+To properly evaluate a detection we must first find the best matches of predictions to ground truths. We start by iterating over predictions by score from highest to lowest. For each prediction we assign the ground truth with the highest IoU value. Both the prediction and ground truth are now considered paired and removed from the pool of choices.
```

Comment:
- comma after "detection"
- iterating over predictions by score from highest to lowest -> iterating over our predictions, starting with the highest scores first
- For each prediction we assign the ground truth with the highest IoU value. -> We pair each prediction with the ground truth that has the highest calculated IOU.